src/registries/menu_items_registry.ts

```diff
@@ -32,30 +32,32 @@ export interface MenuItemSpec {
   id?: string;
   isVisible?: (env: SpreadsheetChildEnv) => boolean;
   isEnabled?: (env: SpreadsheetChildEnv) => boolean;
+  isSelected?: (env: SpreadsheetChildEnv) => boolean;
+  icon?: string;
   isReadonlyAllowed?: boolean;
   action?: (env: SpreadsheetChildEnv) => unknown;
   children?: MenuChildren;
   separator?: boolean;
-  icon?: string;
   textColor?: string;
 }
 
 export interface MenuItem {
-  id: string;
-  sequence: number;
   name: (env: SpreadsheetChildEnv) => string;
+  description: string;
+  sequence: number;
+  id: string;
   isVisible: (env: SpreadsheetChildEnv) => boolean;
   isEnabled: (env: SpreadsheetChildEnv) => boolean;
+  isSelected?: (env: SpreadsheetChildEnv) => boolean;
```

Comment:
Maybe make it mandatory, with a default value `() => false` to make it easier to use/consistent with isVisible/isEnabled

Reply:
Yes and no:
have it optional, allow knowing if we shoud keep a space to accommodate the "check icon".

For example, if number formats of a selected cell does not correspond to the format offered in number format menu:
--> no format will be selected in the menu, 
--> but we want to leave empty spaces to make the user understand that the current format not correspond 

Conversely, there are menus that will never have the opportunity to have menu items selected or with default icons. In these cases, it is useless to have a space

See here: https://github.com/odoo/o-spreadsheet/pull/2046/files#diff-5a985db40b62ab2036f2e38176aaeb51df0a0da1ceef433f81b698d0ba783b6cR164

Reply:
ok to replace ```isSelected``` by ```isActive```